src/activities/morse_code/MorseCode.qml

```diff
@@ -86,8 +86,15 @@ ActivityBase {
             }
 
             function initSubLevel() {
+                if(dataset[currentLevel].values[0] == '_random_') {
+                    var randomIndex = Math.floor(Math.random() * dataset[currentLevel].values[1].length)
+                    questionValue = dataset[currentLevel].values[1][randomIndex]
+                    dataset[currentLevel].values[1].splice(randomIndex,1)
+                    score.numberOfSubLevels = dataset[currentLevel].values[2]
```

Comment:
I'm removing the letters from the array which already appeared by the random selection, so that they do not appear again. Hence the length of the array gets changed so numberOfSubLevels will report decremented value each time. So, the sublevel will change before all letters have appeared as the correct input count catches up with the numberOfSubLevels prematurely.